trio/_util.py

```diff
@@ -252,7 +269,10 @@ def fix_one(qualname, name, obj):
             fix_one(objname, objname, obj)
 
 
-class generic_function:
+# We need ParamSpec to type this "properly", but that requires a runtime typing_extensions import
+# to use as a class base. This is only used at runtime and isn't correct for type checkers anyway,
+# so don't bother.
+class generic_function(t.Generic[RetT]):
```

Comment:
I think a runtime import isn't too bad, but now that I think of it I forgot we don't depend on `typing_extensions` iirc...

Reply:
yeah, we only depend on it for tests.